docs/contributing/coding-style.md

```diff
@@ -517,7 +535,12 @@ func New(opts ...Option) (Server, error) {
     srv := new(server)
     for _, opt := range opts {
         if err := opt(srv); err != nil {
-            return nil, errors.ErrOptionFailed(err, reflect.ValueOf(opt))
+            werr := errors.ErrOptionFailed(err, reflect.ValueOf(opt))
+            log.Warn(werr)
```

Comment:
I think you should error output when an critical error`s type  is not IsCriticalOptionError
Could you please read this ticket`s exampe?
https://github.com/vdaas/vald/projects/3#card-44665812

Reply:
Sorry maybe I am not quite understand your opinion, can you write example code?

Reply:
```
w := new(hoge)
for _, opt := range append(defaultOpts, opts...) {
    if errors. IsCriticalOptionError(err) {
        return nil, err
    }

    log.Warnf(err, errors.OptionAllyFailed)
}

```

Reply:
@hlts2 I think your implementation will not log the error message when it is an critical error. 
My implementation will log the error first and return it if it is critical error.

```go
func New(opts ...Option) (Server, error) {
    srv := new(server)
    for _, opt := range opts {
        if err := opt(srv); err != nil {
            werr := errors.ErrOptionFailed(err, reflect.ValueOf(opt))
            log.Warn(werr)

            if errors.IsCriticalOptionError(err) {
                return nil, werr
            }
        }
    }
}
```

Do you think it is ok?

Reply:
@hlts2 I think the following is better, what do you think?
Your implementation will print out the logging message without the function name

```go
w := new(hoge)
for _, opt := range append(defaultOpts, opts...) {
    werr := errors.ErrOptionFailed(err, reflect.ValueOf(opt))
    if errors. IsCriticalOptionError(err) {
        log.Error(werr)
        return nil, err
    }

    log.Warnf(werr)
}

```

Reply:
it looks good to me